lib/media/drm_engine.js

```diff
@@ -474,6 +475,14 @@ shaka.media.DrmEngine.prototype.prepareMediaKeyConfigs_ =
         variant.drmInfos = [clearKeyDrmInfo];
       }
 
+      // If initial manifest contains unencrypted content,
+      // drm configuration overrides DrmInfo so drmEngine can be activated.
+      // Thus, player can play encrypted content if live stream switches from
+      // unencrypted content to encrypted content during live stream.
+      if (!!configDrmInfos) {
+        variant.drmInfos = configDrmInfos;
```

Comment:
This drmInfos is going to be generated with default drmInfos. When stream switches to encrypted content, encrypted content might have a different configured drmInfos. At this situation, I do not know if there is any risk that may break the playback. As for my tests in our live streams, I could not reproduce any issue.